server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/bucket/RandomSamplerIT.java

```diff
@@ -123,9 +124,12 @@ public void testRandomSamplerConsistentSeed() {
                     ),
                 response -> {
                     InternalRandomSampler sampler = response.getAggregations().get("sampler");
-                    assertThat(((Avg) sampler.getAggregations().get("mean_monotonic")).getValue(), equalTo(sampleMonotonicValue[0]));
-                    assertThat(((Avg) sampler.getAggregations().get("mean_numeric")).getValue(), equalTo(sampleNumericValue[0]));
-                    assertThat(sampler.getDocCount(), equalTo(sampledDocCount[0]));
+                    double monotonicValue = ((Avg) sampler.getAggregations().get("mean_monotonic")).getValue();
+                    double numericValue = ((Avg) sampler.getAggregations().get("mean_numeric")).getValue();
+                    long docCount = sampler.getDocCount();
+                    assertThat(Math.abs(monotonicValue - sampleMonotonicValue[0]) < tolerance, equalTo(true));
```

Comment:
This looks good, but I think it could be a bit better. If we use assertEquals instead of assertThat, I think we will get a slightly better error message if this fails. In junit.Assert there is an equals method for doubles which accepts a tolerance:

```suggestion
                    assertEquals(monotonicValue, sampleMonotonicValue[0], tolerance);
```

I would appreciate it if you would make this change.

Reply:
Thanks for the feedback! I’ll update the PR in the next 30 minutes.